ResearchKit/Common/ORKFormItemCell.m

```diff
@@ -1148,3 +1156,125 @@ - (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRang
+    ORKEnableAutoLayoutForViews([dictionary allValues]);
+    NSDictionary *metrics = @{@"verticalMargin":@(10), @"horizontalMargin":@(self.separatorInset.left), @"verticalMarginBottom":@(10.0 - (1.0 / [UIScreen mainScreen].scale))};
+    
+    [constraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-horizontalMargin-[_selectionView]-horizontalMargin-|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:metrics views:dictionary]];
+    [constraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|-verticalMargin-[_selectionView]-verticalMarginBottom-|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:metrics views:dictionary]];
+    _heightConstraint = [NSLayoutConstraint constraintWithItem:_selectionView attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:nil attribute:NSLayoutAttributeNotAnAttribute multiplier:1.0 constant:_selectionView.intrinsicContentSize.height];
+    _heightConstraint.priority = UILayoutPriorityDefaultHigh;
+    [constraints addObject:_heightConstraint];
+    
+    [self.contentView addConstraints:constraints];
+}
+
+- (void)setFormItem:(ORKFormItem *)formItem {
+    [super setFormItem:formItem];
+    
+    if (_selectionView) {
+        [_selectionView setPlaceholderText:formItem.placeholder];
+    }
+}
+
+- (void)answerDidChange {
+    id answer = self.answer;
+    
+    if (ORKIsAnswerEmpty(answer)) {
+        answer = nil;
+    }
+    
+    _selectionView.answer = (MKPlacemark *)answer;
+    [_selectionView setTextColor:[UIColor blackColor]];
+}
+
+- (void)setEditingHighlight:(BOOL)editingHighlight {
+    _editingHighlight = editingHighlight;
+    [_selectionView setTextColor:( _editingHighlight ? [self tintColor] : [UIColor blackColor])];
+}
+
+- (void)locationSelectionViewDidBeginEditing:(ORKLocationSelectionView *)view {
+    self.editingHighlight = YES;
+    [_selectionView showMapView];
+    [self.delegate formItemCellDidBecomeFirstResponder:self];
+}
+
+- (void)locationSelectionViewDidEndEditing:(ORKLocationSelectionView *)view {
+    self.editingHighlight = NO;
+    [self.delegate formItemCellDidResignFirstResponder:self];
+}
+
+- (void)locationSelectionViewDidChange:(ORKLocationSelectionView *)view {
+    [self inputValueDidChange];
+}
+
+- (void)locationSelectionViewNeedsResize:(ORKLocationSelectionView *)view {
+    UITableView *tableView = [self parentTableView];
+    
+    _heightConstraint.constant = _selectionView.intrinsicContentSize.height;
+    
+    [tableView beginUpdates];
+    
+    CGRect convertedVisibleRect = [tableView convertRect:view.bounds fromView:view];
+    [tableView scrollRectToVisible:convertedVisibleRect animated:YES];
+    
+    [tableView endUpdates];
+}
+
+- (void)locationSelectionView:(ORKLocationSelectionView *)view didFailWithError:(NSError *)error {
+    [self showErrorAlertWithTitle:ORKLocalizedString(@"LOCATION_ERROR_TITLE", @"") message:error.localizedDescription];
```

Comment:
Same here, need to translate `error.localizedDescription` human readable content.